pkg/skaffold/schema/validation/validation.go

```diff
@@ -189,11 +210,60 @@ func validateDockerNetworkMode(artifacts []*latest.Artifact) (errs []error) {
 		if mode == "none" || mode == "bridge" || mode == "host" {
 			continue
 		}
+		containerRegExp := regexp.MustCompile("^container:[a-zA-Z0-9][a-zA-Z0-9_.-]*$")
+		if containerRegExp.MatchString(mode) {
+			continue
+		}
+
 		errs = append(errs, fmt.Errorf("artifact %s has invalid networkMode '%s'", a.ImageName, mode))
 	}
 	return
 }
 
+// Validates that a Docker Container with a Network Mode "container:<id|name>" points to an actually running container
+func validateDockerNetworkContainerExists(artifacts []*latest.Artifact, runCtx docker.Config) []error {
+	var errs []error
+	apiClient, err := docker.NewAPIClient(runCtx)
+	if err != nil {
+		errs = append(errs, err)
+		return errs
+	}
+
+	client := apiClient.RawClient()
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(5*time.Second))
```

Comment:
Thanks @zedfmario for adding this validating. Is 5 seconds reasonable? Sorry, i have never used docker container runtimes.

Would you up for using skaffold's new ActionableError proto to add error codes and suggestions?

In order to add error codes, 
1. Add a Distinct Err Code in proto/skaffold.proto e.g. `INIT_INVALID_DOCKER_NETWORK_CONTAINER_NAME`, 
`INIT_DOCKER_NETWORK_CONTAINER_DOES_NOT_EXIST` and Suggestion codes like `FIX_DOCKER_NETWORK_CONTAINER_NAME`, `CHECK_DOCKER_NETWORK_CONTAINER_RUNNING`
Here "INIT" stands for the phase initialization.
2. run `./hack/generate_proto.sh`
3. You can use the new [`sError.NewError`](https://github.com/GoogleContainerTools/skaffold/blob/54466ff6983e9fcf977d6e549119b4c1c4dc9e2b/pkg/skaffold/errors/err_def.go#L57) to create errors with the distinct error code you just defined e.g
```
return sErrors.NewError(err,
		proto.ActionableErr{
			Message: fmt.Sprintf("artifact %s has invalid networkMode '%s'", imageName, mode),
			ErrCode: proto.StatusCode_INIT_INVALID_DOCKER_NETWORK_CONTAINER_NAME,
                         Suggestions: []*proto.Suggestion{
				{
					SuggestionCode: proto.SuggestionCode_FIX_DOCKER_NETWORK_CONTAINER_NAME,
					Action:         "Please fix the docker network container name and try again.",
				},
			},
		})
```
4. Finally,  at L81, you can just return errs.

Sorry, this is a lot more work. Would really appreciate if you could add it.

Reply:
Hi! thank you for your hints! I'm gonna adapt it ASAP.

Regarding 5 seconds being reasonable, Docker usually responds faster than that and the context will be "Done". Anyways, I'm not an expert in the subject either. I assume that 5 seconds should be enough. I've never faced longer times but I'm OK with increasing it if you feel like that'd be a better approach...

Reply:
Hi again,

Sorry for asking. Regarding the comment _Finally, at L81, you can just return errs_, the function `ProcessWithRunContext` returns a single `error`. What should I do here? I was following the same structure as `Process`, appending all possible errors...

Reply:
Thanks. appending all errors is good for now.  I have to do some work to the error infrastructure to solve that issue. 
I don't have an good answer on how to solve it right now.